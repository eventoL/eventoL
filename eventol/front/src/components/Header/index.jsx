import React from 'react';
import PropTypes from 'prop-types';

import Logo from '../Logo';
import Navigation from '../Navigation';
import UserProfile from '../UserProfile';
import SignIn from '../SignIn';

import './index.scss';
import {LOGO_HEADER_DEFAULT} from '../../utils/constants';

export default class Header extends React.PureComponent {
  static propTypes = {
    handlerOnChangeLanguage: PropTypes.func.isRequired,
    isMobile: PropTypes.bool.isRequired,
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
      })
    ),
    logoHeader: PropTypes.string,
    user: PropTypes.shape({
      first_name: PropTypes.string,
      last_name: PropTypes.string,
    }),
  };

  static defaultProps = {
    languages: [],
    logoHeader: LOGO_HEADER_DEFAULT,
    user: null,
  };

  handleToggle = event => {
    event.preventDefault();
    event.stopPropagation();
    const linksEl = document.querySelector('.narrow-links');
    const {
      style: {display},
    } = linksEl;
    linksEl.style.display = display === 'block' ? 'none' : 'block';
  };

  showUserIndicator = () => {
    const {user} = this.props;
    if (user) return <UserProfile user={user} />;
    return <SignIn />;
  };

  wideRender = () => {
    const {logoHeader, languages, handlerOnChangeLanguage} = this.props;
    return (
      <header className="header">
        <div className="nav-wide">
          <Logo logoHeader={logoHeader} />
          <Navigation
            handlerOnChangeLanguage={handlerOnChangeLanguage}
            languages={languages}
          />
          {this.showUserIndicator()}
        </div>
      </header>
    );
  };

  render() {
    const {
      isMobile,
      logoHeader,
      languages,
      handlerOnChangeLanguage,
    } = this.props;
    if (!isMobile) return this.wideRender();
    return (
      <header className="header">
        <div className="nav-narrow">
          <Logo logoHeader={logoHeader} />
          <div className="nav-right">
            <button
              onClick={this.handleToggle}
              onKeyPress={this.handleToggle}
              type="button"
            >
              <i className="fa fa-bars fa-2x" />
            </button>
            <div className="narrow-links">
              <Navigation
                handlerOnChangeLanguage={handlerOnChangeLanguage}
                languages={languages}
              />
              {this.showUserIndicator()}
            </div>
          </div>
        </div>
      </header>
    );
  }
}
