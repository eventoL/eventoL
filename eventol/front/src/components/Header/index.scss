@import '../../styles/index';

.header {
  @extend %flex-element;
  // FF3.6-15
  background: -webkit-linear-gradient(top, $gradient-color 0%, transparent 100%);

  // Chrome10-25, Safari5.1-6
  background: linear-gradient(to bottom, $gradient-color 0%, transparent 100%);

  box-sizing: border-box;

  // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#a6000000', endColorstr='#00000000', GradientType=0);

  left: 0;
  padding: 40px;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 6;
}

.nav-wide {
  display: none;
}

@media (min-width: 950px) {
  .nav-wide {
    @include flex-element;
    box-sizing: border-box;
    width: 100vw;
  }

  .nav-narrow {
    display: none;
  }
}

@media (max-width: 950px) {
  .logo-component {
    position: absolute;
  }
}

.nav-narrow {
  @extend %flex-element;
  box-sizing: border-box;
  left: 0;
  padding: 40px;
  position: fixed;
  top: 0;
  width: 100vw;

  .nav-right {
    margin-left: auto;
    z-index: 7;

    button {
      background-color: transparent;
      border: 0;
      color: $white;
      display: contents;
      font-size: 100%;
      margin: 0;
      padding: 0;
      vertical-align: baseline;
    }

    i {
      float: right;
      z-index: 5;
    }

    .narrow-links {
      display: none;

      a {
        clear: right;
        display: inline;
      }
    }
  }
}

a {
  color: $a-color;
  text-decoration: none;

  &:focus,
  &:hover {
    color: $a-color;
    text-decoration: none;
  }
}
