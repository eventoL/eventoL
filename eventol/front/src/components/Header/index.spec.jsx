import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../Logo', () => 'Logo');
jest.mock('../Navigation', () => 'Navigation');
jest.mock('../UserProfile', () => 'UserProfile');
jest.mock('../SignIn', () => 'SignIn');

import Header from '.';

describe('Header', () => {
  let component, tree, user, logo, isMobile;

  beforeEach(() => {
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    logo = 'eventol.png';
  });

  describe('With user', () => {
    describe('Mobile version', () => {
      beforeEach(() => {
        isMobile = true;
        component = renderer.create(
          <Header logoHeader={logo} user={user} isMobile={isMobile} />,
        );
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('Desktop version', () => {
      beforeEach(() => {
        isMobile = false;
        component = renderer.create(
          <Header logoHeader={logo} user={user} isMobile={isMobile} />,
        );
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('Without user and without logo', () => {
    describe('Mobile version', () => {
      beforeEach(() => {
        isMobile = true;
        component = renderer.create(
          <Header isMobile={isMobile} />,
        );
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('Desktop version', () => {
      beforeEach(() => {
        isMobile = false;
        component = renderer.create(
          <Header isMobile={isMobile} />,
        );
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
