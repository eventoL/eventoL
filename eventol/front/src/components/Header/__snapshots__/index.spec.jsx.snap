// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header With user Desktop version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-wide"
  >
    <Logo
      logoHeader="eventol.png"
    />
    <Navigation />
    <UserProfile
      user={
        Object {
          "first_name": "first_name",
          "last_name": "last_name",
        }
      }
    />
  </div>
</header>
`;

exports[`Header With user Mobile version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-narrow"
  >
    <Logo
      logoHeader="eventol.png"
    />
    <div
      className="nav-right"
    >
      <button
        onClick={[Function]}
        onKeyPress={[Function]}
        type="button"
      >
        <i
          className="fa fa-bars fa-2x"
        />
      </button>
      <div
        className="narrow-links"
      >
        <Navigation />
        <UserProfile
          user={
            Object {
              "first_name": "first_name",
              "last_name": "last_name",
            }
          }
        />
      </div>
    </div>
  </div>
</header>
`;

exports[`Header Without user and without logo Desktop version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-wide"
  >
    <Logo
      logoHeader="/static/manager/img/eventol-white.png"
    />
    <Navigation />
    <SignIn />
  </div>
</header>
`;

exports[`Header Without user and without logo Mobile version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-narrow"
  >
    <Logo
      logoHeader="/static/manager/img/eventol-white.png"
    />
    <div
      className="nav-right"
    >
      <button
        onClick={[Function]}
        onKeyPress={[Function]}
        type="button"
      >
        <i
          className="fa fa-bars fa-2x"
        />
      </button>
      <div
        className="narrow-links"
      >
        <Navigation />
        <SignIn />
      </div>
    </div>
  </div>
</header>
`;
